src/ReDate.re

```diff
@@ -676,3 +676,102 @@ let eachDayOfIntervalList = interval =>
       interval |> makeIntervalDay,
     )
   );
+
+/* ——[ISO Week helpers]——————— */
+
+let endOfISOWeek = date => endOfWeek(~weekStartsOn=Monday, date);
+
+let isSameISOWeek = (fst, snd) => isSameWeek(~weekStartsOn=Monday, fst, snd);
+
+let isThisISOWeek = date => isSameISOWeek(Js.Date.make(), date);
+
+let lastDayOfISOWeek = date => lastDayOfWeek(~weekStartsOn=Monday, date);
+
+let startOfISOWeek = date => startOfWeek(~weekStartsOn=Monday, date);
+
+let differenceInCalendarISOWeeks = (fst, snd) => {
+  let startOfFstISOWeek = startOfISOWeek(fst);
+  let startOfSndISOWeek = startOfISOWeek(snd);
+
+  let timestampFst =
+    Js.Date.getTime(startOfFstISOWeek)
+    -. Internal.getTimezoneOffsetInMilliseconds(startOfFstISOWeek);
+  let timestampSnd =
+    Js.Date.getTime(startOfSndISOWeek)
+    -. Internal.getTimezoneOffsetInMilliseconds(startOfSndISOWeek);
+
+  let milliseconds_in_week = 604800000.;
+  (timestampFst -. timestampSnd)
+  /. milliseconds_in_week
+  |> Js.Math.round
+  |> int_of_float;
+};
+
+/* ——[ISO Week-Numbering Year Helpers]—— */
+
+let getISOWeekYear = date => {
+  let year = Js.Date.getFullYear(date);
+
+  let fourthOfJanuaryOfNextYear =
+    Js.Date.setFullYearMD(
+      Js.Date.fromFloat(0.),
+      ~year=year +. 1.,
+      ~month=0.,
+      ~date=4.,
+      (),
+    )
+    |> Js.Date.fromFloat
+    |> Internal.makeDateWithStartOfDayHours;
+  let startOfNextYear = fourthOfJanuaryOfNextYear |> startOfISOWeek;
+
+  let fourthOfJanuaryOfThisYear =
+    Js.Date.setFullYear(
+      fourthOfJanuaryOfNextYear |> Js.Date.getTime |> Js.Date.fromFloat,
+      year,
+    )
+    |> Js.Date.fromFloat;
+  let startOfThisYear = fourthOfJanuaryOfThisYear |> startOfISOWeek;
+
+  if (Js.Date.getTime(date) >= Js.Date.getTime(startOfNextYear)) {
+    year +. 1. |> int_of_float;
+  } else if (Js.Date.getTime(date) >= Js.Date.getTime(startOfThisYear)) {
+    year |> int_of_float;
+  } else {
+    year -. 1. |> int_of_float;
+  };
+};
+
+let startOfISOWeekYear = date => {
+  let year = getISOWeekYear(date) |> float_of_int;
+
+  let fourthOfJanuary = Js.Date.fromFloat(0.);
+  let fourthOfJanuary =
+    Js.Date.setFullYearMD(fourthOfJanuary, ~year, ~month=0., ~date=4., ())
+    |> Js.Date.fromFloat
+    |> Internal.makeDateWithStartOfDayHours;
+  let startOfThisYear = startOfISOWeek(fourthOfJanuary);
+  startOfThisYear;
+};
+
+let getISOWeek = date => {
```

Comment:
Because `getISOWeek` need to access `startOfISOWeekYear`, it have been taken out of it's section and placed down here, is this OK or is it other ways to do this?

Reply:
it's definitely ok!